### ISSUE TITLE
Descuento Global en % no calcula valores correctamente y no sirve para analisis

### ISSUE THREAD

**Issue by celm1990** [dansanti/global_discount#2]
Cuando en una factura se aplica un descuento global en %, este % no se pasa a las lineas sino que se implementa un calculo medio raro para que funcione, pero realmente no funciona bien cuando el impuesto esta incluido en el precio, ya que para el calculo se usa la columna price_subtotal de las lineas de factura y esa columna resta el impuesto cuando este esta incluido en el precio. Esto causa que el total de la factura sea errado. Ademas usando descuentos de esta manera, en los analisis de factura no se aprecia el descuento y sale el total de la orden sin descuento.

Sugieron que cuando sea descuento global en %, lo que se haga es pasar ese % a cada linea, asi el sistema calculara de forma natural el descuento.

Como dato adicional,  la funcion _compute_amount de account.invoice tienen unos decoradores especificos para calcular los campos funcionales en funcion de otros campos(depends), pero en este modulo se reemplaza la funcion y se cambian los decoradores por onchange, lo que provoca que los campos calculados no se disparen correctamente y se recalculen los datos.(cuando ya tengo ingresadas las lineas y guardo la factura, luego edito la factura y pongo un descuento, los totales no se recalculan)

**DESCUENTO NORMAL**
![image](https://cloud.githubusercontent.com/assets/7775116/26223285/1cf23cb6-3be3-11e7-8b4d-92c06d0ff6f1.png)

**DESCUENTO GLOBAL**
![image](https://cloud.githubusercontent.com/assets/7775116/26223307/2e6f933a-3be3-11e7-92a3-916b8212507e.png)

El resultado deberia ser el mismo, pero al poner el descuento global, el total es diferente(en este caso los precios incluyen impuestos, asi esta configurado en el impuesto)

**Comment by dansanti** [comment]
hola!
no lo he probado con iva incluido.

- Por normativa del sii debe ser calculado sobre el monto neto tal como está ahí, 
- El descuento global difiere al descuento por línea, hacer eso , estaríamos no cumpliendo con los requerimientos mínimos del SII, para pasar la certificación Electrónica.
- Los resumenes no lo he visto, pero debieran quedar bien (almenos con iva no incluido) ya que el valor neto y total, se guarda en la db

**Comment by nelsonramirezs** [comment]
Hola, ceelm1999 tiene razón. 
Los puntos de ambos son válidos, pero el cálculo no da. En la foto ceelm muestra que un artículo que cuesta 30.000 le haces 50% de descuento y no da 15.000 cuando este incluye el iva. La normativa dice que al neto pero en este caso da igual el resultado. 
Está casi perfecto pero el impuesto debe ser 2395.
(30.000 / 1.19) *(50 / 100) * 0.19

Haré fork

**Comment by dansanti** [comment]
Claro, cuando es incluído hay que revisarlo, porque manualment sin iva incluído , tu ejemplo me cuadra:

$30.000/1,19
$ 25.210
Global Discount Amount
$ 12.605
Impuesto
$ 2.395
Total
$ 15.000

Puede que esta semana le dedique un tiempo, pero estoy con lo de webpay aún

**Comment by dansanti** [comment]
Hola por favor verificar si repara este issue el último commit, ya que reestructuré casi todo y agregué más opciones, como por ejemplo agregar multiples descuentos y la posibilidad de agregar recargos https://github.com/dansanti/global_discount/commit/c90c3011d980bb3dabcf58847a9e1e137289368d

**Comment by celm1990** [comment]
Hola, estuve probando este tema pero aun veo problemas con los descuentos globales:
1.- no hay soporte para pedidos de compra y venta, solo facturas
2.- no es una distribución en las lineas, lo que hace imposible ver el descuento en los analisis que se basan en las lineas(analisis de factura)

Por ahora es lo que he podido notar, sigo probando y te aviso, pero seria importante replicar el concepto a pedidos de compra y venta para q todo haga sentido

**Comment by dansanti** [comment]
No esyn descuento por lineas, revisa la documentación del sii para entender
lo que es un descuento o recargo global

**Comment by dansanti** [comment]
ya entendí el problema, lo que requieren es que el descuento se haga sobre el bruto, no sobre el neto, actualmente el módulo hace descuento sobre el neto
